pypsrp/wsman.py

```diff
@@ -773,6 +783,21 @@ def _build_session(self):
         elif self.no_proxy:
             session.proxies = orig_proxy
 
+        # Retry on connection errors, with a backoff factor
+        retries = requests.packages.urllib3.util.retry.Retry(
+            total=self.reconnection_retries,
+            connect=self.reconnection_retries,
+            read=self.reconnection_retries,
+            backoff_factor=self.reconnection_backoff,
+            status_forcelist=[425, 429, 503],
+        )
+        session.mount('http://', requests.adapters.HTTPAdapter(
```

Comment:
We should only add based on the prefix that is being used right now, use `scheme = 'https' if self.ssl else 'http'` to get the scheme.

Reply:
I don't think it really matters that much, but we can.

Reply:
It does matter, since redirects can go from http to https (insecure), or vice-versa (with or without proxying) I think we need to handle both scenarios. If you don't define both, urllib3 will not retry if this happens.

Reply:
Ok